primer/src/Primer/TypeDef.hs

```diff
@@ -105,6 +105,9 @@ typeDefAST = \case
 typeDefKind :: TypeDef b () -> Kind' ()
 typeDefKind = foldr (KFun () . snd) (KType ()) . typeDefParameters
 
+-- TODO/REVIEW: I cannot see how to do this nice and lens-y (see old code
+-- above) because we cannot change metadata in params indep to in ctors,
+-- as both controlled by same `c`
```

Comment:
TODO: I'll rebase this commit away after reviews are done (it has been migrated to a review comment)